lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -0,0 +1,533 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: aws_ec2
+    plugin_type: inventory
+    short_description: ec2 inventory source
+    extends_documentation_fragment:
+        - inventory_cache
+        - constructed
+    description:
+        - Get inventory hosts from Amazon Web Services EC2.
+        - Uses a <name>.aws_ec2.yaml (or <name>.aws_ec2.yml) YAML configuration file.
+    options:
+        boto_profile:
+          description: The boto profile to use.
+          env:
+              - name: AWS_PROFILE
+              - name: AWS_DEFAULT_PROFILE
+        aws_access_key_id:
+          description: The AWS access key to use. If you have specified a profile, you don't need to provide
+              an access key/secret key/session token.
+          env:
+              - name: AWS_ACCESS_KEY_ID
+              - name: AWS_ACCESS_KEY
+              - name: EC2_ACCESS_KEY
+        aws_secret_access_key:
+          description: The AWS secret key that corresponds to the access key. If you have specified a profile,
+              you don't need to provide an access key/secret key/session token.
+          env:
+              - name: AWS_SECRET_ACCESS_KEY
+              - name: AWS_SECRET_KEY
+              - name: EC2_SECRET_KEY
+        aws_security_token:
+          description: The AWS security token if using temporary access and secret keys.
+          env:
+              - name: AWS_SECURITY_TOKEN
+              - name: AWS_SESSION_TOKEN
+              - name: EC2_SECURITY_TOKEN
+        regions:
+          description: A list of regions in which to describe EC2 instances. By default this is all regions except us-gov-west-1
+              and cn-north-1.
+        hostnames:
+          description: A list in order of precedence for hostname variables. You can use the options specified in
+              U(http://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html#options). To use tags as hostnames
+              use the syntax tag:Name=Value to use the hostname Name_Value.
+        filters:
+          description: A dictionary of filter value pairs. Available filters are listed here
+              U(http://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html#options)
+        strict_permissions:
+          description: By default if a 403 (Forbidden) is encountered this plugin will fail. You can set strict_permissions to
+              False in the inventory config file which will allow 403 errors to be gracefully skipped.
+'''
+
+EXAMPLES = '''
+simple_config_file:
+
+    plugin: aws_ec2
+    boto_profile: aws_profile
+    regions: # populate inventory with instances in these regions
+      - us-east-1
+      - us-east-2
+    filters:
+    # filter by tags with the value dev
+      tag:Name: dev
```

Comment:
This looks for tags with the `Name` key set to `dev` - maybe a better example would be:

```
# all instances with their `Environment` tag set to `dev`
tag:Environment: dev
# all dev and QA hosts
tag:Environment:
- dev
- qa
```
Since `tag:Name` could be confused for "tags with the name dev" as in `dev : true` or something

Reply:
I will follow this up with a docs fix.